server/rest/postgres_geojson.py

```diff
@@ -0,0 +1,368 @@
+        .param('assetstoreId', 'assetstore ID of the target database')
+        .param('table', 'Table name from the database')
+        .param('column', 'Column name from a table')
+    )
+    def getValues(self, assetstore, params):
+        return self._getValues(assetstore, params)
+
+    def _getValues(self, assetstore, params):
+        adapter = assetstore_utilities.getAssetstoreAdapter(assetstore)
+        conn = adapter.getDBConnectorForTable(params['table'])
+        queryParams = {
+            'fields': [{
+                'func': 'distinct',
+                'param': [{'field': params['column']}],
+                'reference': 'value',
+            }],
+            'limit': 100,
+            'format': 'rawdict'}
+        result = list(adapter.queryDatabase(conn, queryParams)[0]())
+        return [row['value'] for row in result]
+
+    @access.user
+    @loadmodel(model='assetstore', map={'assetstoreId': 'assetstore'})
+    @describeRoute(
+        Description('Returns all distinct values for all columns for a given table')
+        .param('assetstoreId', 'assetstore ID of the target database')
+        .param('table', 'Table name from the database')
+    )
+    def getAllValues(self, assetstore, params):
+        resp = {}
+        for i in self._getColumns(assetstore, params):
+            if i['name'] != 'geom' and i['datatype'] != 'number':
+                resp[i['name']] = self._getValues(assetstore, {
+                    'column': i['name'],
+                    'table': params['table']})
+        return resp
+
+    @access.user
+    @loadmodel(model='assetstore', map={'assetstoreId': 'assetstore'})
+    @describeRoute(
+        Description('Create json dataset for the given view/table filtering values')
+        .param('assetstoreId', 'assetstore ID of the target database')
+        .param('table', 'Table name from the database')
+        .param('field', 'Field to which the aggregate function will be applied')
+        .param('aggregationFunction', 'aggregate function used on the field')
+        .param('filter', 'Filter condition object for filtering table data')
+        .param('geometryField', 'Geometry data definition object')
+        .param('datasetName', 'A custom name for the dataset', required=False)
+    )
+    def createPostgresGeojsonData(self, assetstore, params):
+        filter = params['filter']
+        table = params['table']
+        field = params['field']
+        aggregateFunction = params['aggregateFunction']
+        geometryField = json.loads(params['geometryField'])
+
+        if geometryField['type'] == 'built-in':
+            buildInGeomField = geometryField['field']
+            properties = [field, {
+                'func': aggregateFunction,
+                'param': {'field': field}
+            }]
+            # add string fields with concat aggregate function and in the format for
+            # json_build_object
+            for property in [(i['name'], {
```

Comment:
Can we make this more readable? Maybe extract some of the expressions to separate variables?